app/mailers/notification_mailer.rb

```diff
@@ -76,17 +76,23 @@ def set_account
   end
 
   def set_list_headers!
-    headers['List-ID'] = "<#{@type}.#{@me.username}.#{Rails.configuration.x.local_domain}>"
-    headers['List-Unsubscribe'] = "<#{@unsubscribe_url}>"
-    headers['List-Unsubscribe-Post'] = 'List-Unsubscribe=One-Click'
+    headers(
+      'List-ID' => "<#{@type}.#{@me.username}.#{Rails.configuration.x.local_domain}>",
+      'List-Unsubscribe-Post' => 'List-Unsubscribe=One-Click',
+      'List-Unsubscribe' => "<#{@unsubscribe_url}>"
+    )
   end
 
-  def thread_by_conversation(conversation)
-    return if conversation.nil?
+  def thread_by_conversation!
+    return if @status.conversation.nil?
 
-    msg_id = "<conversation-#{conversation.id}.#{conversation.created_at.strftime('%Y-%m-%d')}@#{Rails.configuration.x.local_domain}>"
+    headers(
+      'In-Reply-To' => conversation_message_id,
+      'References' => conversation_message_id
+    )
+  end
 
-    headers['In-Reply-To'] = msg_id
-    headers['References']  = msg_id
+  def conversation_message_id
+    "<#{@status.conversation.to_message_id}@#{Rails.configuration.x.local_domain}>"
   end
```

Comment:
I'm not sure about:
- moving `msg_id` to a method, it's only ever used in this place
- moving `to_message_id` to the `Conversation` model, it's only ever relevant for mail headers, a context that is lost in the `Conversation` class

Reply:
I think I would prefer that, yes.